src/dsp-services.js

```diff
@@ -0,0 +1,170 @@
+        for (const lang of temp['rdfs:label']){
+            prop['label'][lang['@language']] = lang['@value'];
+        }
+    }
+    return props;
+}
+/*
+Array of resnames (strings)
+ */
+export async function getLabelsForResources(resources){
+    let toReturn = [];
+    for (const res of resources){
+        let toAdd = {'ResName': res};
+        toAdd['label'] = {};
+        const temp = await getResByName(res);
+        for (const lang of temp['rdfs:label']){
+            toAdd['label'][lang['@language']] = lang['@value'];
+        }
+        toReturn.push(toAdd);
+    }
+    console.log(toReturn);
+    return toReturn;
+}
+export async function getAllResourcesWithLabels(){
+    let resources = await getAllResourceNames();
+    return await getLabelsForResources(resources);
+}
+export async function getAllResourceNames(){
+    if (!ontology){
+        ontology = await getOntology();
+    }
+    let toReturn = [];
+    for (const o of ontology){
+        if (o.hasOwnProperty('knora-api:isResourceClass') && o['knora-api:isResourceClass']){
+            toReturn.push(o['@id']);
+        }
+    }
+    return toReturn;
+}
+export async function getPropByName(name){
+    if (!ontology){
+        ontology = await getOntology();
+    }
+    if (name.indexOf(ontologyIri) === -1){
+        name = ontologyIri + ':' + name;
+    }
+    for (const o of ontology) {
+        if (o.hasOwnProperty('knora-api:isResourceProperty') && o['knora-api:isResourceProperty']) {
+            if (o['@id'] === name){
+                return o;
+            }
+        }
+    }
+    console.log('Didnt find property with name ---' + name + '---');
+    return null;
+}
+export async function getResByName(name){
+    if (!ontology){
+        ontology = await getOntology();
+    }
+    if (name.indexOf(ontologyIri) === -1){
+        name = ontologyIri + ':' + name;
+    }
+    for (const o of ontology){
+        if (o.hasOwnProperty('knora-api:isResourceClass') && o['knora-api:isResourceClass']){
+            if (o['@id'] === name){
+                return o;
+            }
+        }
+    }
+    console.log('Didnt find resource with name ---' + name + '---');
+    return null;
+}
+export async function getListByIri(iri){
+    let toReturn = [];
+    const res = await fetch('https://' + server + '/v2/lists/' + encodeURIComponent(iri) + '?allLanguages=true' , {
+        method: 'GET'
+    })
+    const json = await res.json();
+    for (const node of json['knora-api:hasSubListNode']){
+        toReturn.push(new ListNode(node['rdfs:label'], node['@id']));
+    }
+    return toReturn;
+}
+export async function getListIriByPropName(name){
+    const prop = await getPropByName(name);
+    return prop['salsah-gui:guiAttribute'].replace('hlist=', '').slice(1, -1); //TODO: This might be unsafe for other projects!
+}
+export async function getListByPropName(name){
+  const iri = await getListIriByPropName(name);
+  return await getListByIri(iri);
+}
```

Comment:
Due to readability leave an empty line between functions

Reply:
Added